SmbIO doesn't work when SortedBucketTransforms have SideInputs

Reproduction: https://github.com/spotify/scio/compare/smb-io-side-input?expand=1

Produces the error:

```
Caused by: java.lang.IllegalArgumentException: calling sideInput() with unknown view
	at org.apache.beam.repackaged.direct_java.runners.core.SimpleDoFnRunner.sideInput(SimpleDoFnRunner.java:244)
	at org.apache.beam.repackaged.direct_java.runners.core.SimpleDoFnRunner.access$1200(SimpleDoFnRunner.java:85)
	at org.apache.beam.repackaged.direct_java.runners.core.SimpleDoFnRunner$DoFnProcessContext.sideInput(SimpleDoFnRunner.java:400)
	at com.spotify.scio.values.SingletonSideInput.get(SideInput.scala:104)
	at com.spotify.scio.values.SideInput.getCache(SideInput.scala:47)
	at com.spotify.scio.values.SideInput.getCache$(SideInput.scala:41)
	at com.spotify.scio.values.SingletonSideInput.getCache(SideInput.scala:102)
	at com.spotify.scio.values.SideInputContext.apply(SideInput.scala:150)
	at com.spotify.scio.smb.SmbTransformWithSideInputsJob$$anonfun$main$3.apply(SmbIOTest.scala:135)
	at com.spotify.scio.smb.SmbTransformWithSideInputsJob$$anonfun$main$3.apply(SmbIOTest.scala:133)
	at com.spotify.scio.smb.SortMergeTransform$ViaTransformWithSideOutput.processElement(SortMergeTransform.scala:243)
```

--- a/scio-smb/src/main/scala/com/spotify/scio/smb/SortMergeTransform.scala
+++ b/scio-smb/src/main/scala/com/spotify/scio/smb/SortMergeTransform.scala
@@ -28,7 +28,7 @@ import org.apache.beam.sdk.extensions.smb.SortedBucketIO.{
   Transformable
 }
 import org.apache.beam.sdk.extensions.smb.SortedBucketTransform.{BucketItem, MergedBucket}
-import org.apache.beam.sdk.transforms.DoFn
+import org.apache.beam.sdk.transforms.{DoFn, ParDo}
 import org.apache.beam.sdk.transforms.DoFn.{Element, OutputReceiver, ProcessElement}
 import org.apache.beam.sdk.transforms.join.CoGbkResult
 import org.apache.beam.sdk.transforms.windowing.BoundedWindow
@@ -127,7 +127,7 @@ object SortMergeTransform {
     override def withSideInputs(
       sides: SideInput[_]*
     ): WithSideInputsWriteBuilder[KeyType, R, W] =
-      new WithSideInputsWriteBuilderTest(sc, read, output)
+      new WithSideInputsWriteBuilderTest(sc, read, output, sides)
 
     override def via(
       transformFn: (KeyType, R, SortedBucketTransform.SerializableConsumer[W]) => Unit
@@ -206,7 +206,8 @@ object SortMergeTransform {
   private[smb] class WithSideInputsWriteBuilderTest[KeyType, K1, K2, R, W: Coder](
     @transient private val sc: ScioContext,
     read: => SCollection[(KeyType, R)],
-    output: TransformOutput[K1, K2, W]
+    output: TransformOutput[K1, K2, W],
+    sides: Seq[SideInput[_]]
   ) extends WithSideInputsWriteBuilder[KeyType, R, W] {
     override def via(
       transformFn: (
@@ -216,7 +217,11 @@ object SortMergeTransform {
         SortedBucketTransform.SerializableConsumer[W]
       ) => Unit
     ): ClosedTap[W] = {
-      val data = read.parDo(new ViaTransformWithSideOutput(transformFn))
+      val data = read.applyTransform(
+        ParDo
+          .of(new ViaTransformWithSideOutput(transformFn))
+          .withSideInputs(sides.map(_.view).asJava)
+      )
       TestDataManager.getOutput(sc.testId.get)(SortedBucketIOUtil.testId(output))
       ClosedTap(TapOf[W].saveForTest(data))
     }

--- a/scio-smb/src/test/scala/com/spotify/scio/smb/SmbIOTest.scala
+++ b/scio-smb/src/test/scala/com/spotify/scio/smb/SmbIOTest.scala
@@ -118,6 +118,29 @@ object SmbTransformJob extends SmbJob {
 
 }
 
+object SmbTransformWithSideInputsJob extends SmbJob {
+
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+    val side = sc.parallelize(List(7)).asSingletonSideInput
+
+    sc.sortMergeTransform(
+      keyClass,
+      avroUsers(args),
+      avroAccounts(args)
+    ).to(avroTransformOutput(args))
+      .withSideInputs(side)
+      .via { case (_, (users, accounts), ctx, outputCollector) =>
+        val user = setUserAccounts(users, accounts)
+        user.setId(ctx(side))
+        outputCollector.accept(user)
+      }
+
+    sc.run().waitUntilDone()
+  }
+
+}
+
 class SmbIOTest extends PipelineSpec {
   private val accountA =
     new Account(1, "typeA", "nameA", 12.5, null)
@@ -188,6 +211,23 @@ class SmbIOTest extends PipelineSpec {
       .input(SmbIO[Integer, Account]("gs://accounts", _.getId), Seq(accountA, accountB))
       .output(SmbIO[Integer, User]("gs://output", _.getId))(
         _ should containInAnyOrder(Seq(joinedUserAccounts))
+      )
+      .run()
+  }
+
+  it should "be able to mock sortMergeTransform with side inputs" in {
+    JobTest[SmbTransformWithSideInputsJob.type]
+      .args(
+        "--users=gs://users",
+        "--accounts=gs://accounts",
+        "--output=gs://output"
+      )
+      .input(SmbIO[Integer, User]("gs://users", _.getId), Seq(user))
+      .input(SmbIO[Integer, Account]("gs://accounts", _.getId), Seq(accountA, accountB))
+      .output(SmbIO[Integer, User]("gs://output", _.getId))(
+        _ should containInAnyOrder(
+          Seq(User.newBuilder(user).setAccounts(List(accountA, accountB).asJava).setId(7).build())
+        )
       )
       .run()
   }

